src/builders.ts

```diff
@@ -1,19 +1,21 @@
+import { relative } from 'path';
 import { todoFilePathFor } from './io';
 import { FilePath, LintMessage, LintResult, TodoData } from './types';
 
 /**
  * Adapts a list of {@link https://github.com/DefinitelyTyped/DefinitelyTyped/blob/160f43ae6852c4eefec2641e54cff96dd7b63488/types/eslint/index.d.ts#L640 ESLint.LintResult}
  * or {@link TemplateLintResult} to a map of {@link FilePath}, {@link TodoData}.
  *
+ * @param baseDir The base directory that contains the .lint-todo storage directory.
  * @param lintResults A list of {LintResult} objects to convert to {TodoData} objects.
  */
-export function buildTodoData(lintResults: LintResult[]): Map<FilePath, TodoData> {
+export function buildTodoData(baseDir: string, lintResults: LintResult[]): Map<FilePath, TodoData> {
```

Comment:
Can we make these all relative to the storage dir (instead of requiring both storage dir and base dir in some contexts), what do you think?

Reply:
We can do anything we want :). It does seem a bit odd to make it relative to the storage directory mainly because we're planning on allowing you to set the storage directory yourself. 

The problem I see is that the results are project based, not storage directory based. This means that files we lint are relative to the base directory of the project being linted, and the storage directory doesn't really mean anything in this case - it's just meta data. It feels a little odd to modify the internal `filePath` to be different than that of the base directory. Put another way, the original lint violation will contain the base directory portion _and_ the relative portion to the file, but that's relative to the base directory, and nothing else.

What do you think?

Reply:
Yeah, I definitely get what you mean here. I guess my concern is around portability. Taking multiple stand alone repos (where `.lint-todo` is in the root, and files are relative to repo root) and merging them into a mono-repo setup (where `packages/some-name/.lint-todo` and files are in `packages/some-name/whatever/path/here.hbs`) will result in additional knock on changes being required in the `.lint-todo` files (therefore complicating the migration to a monorepo in a way that is not very clear).

If the `TodoData` are relative to the storage directory, then as long as the relationship between the storage dir and the file being linted does not change, the files themselves can be moved around at will and still be stable.

Reply:
So to be clear, the reason I'm passing `baseDir` around is to correctly strip it off the lint results from both `eslint` and `ember-template-lint`, as those libs return absolute paths in the results. We need to pass this in to be able to generate the relative path from the absolutes. I'm sure you understand this already, but wanted to make sure we're 100% talking about the same thing.

If we want to further make the the todos relative to the storage directory, that seems fine to do.

Reply:
Would you be OK with merging this in now, and us address this in a follow up later (before this lib is in use?)

Reply:
> Would you be OK with merging this in now, and us address this in a follow up later (before this lib is in use?)

Yep! My concern isn't really blocking per-se, I just want to figure out the right balance...

Reply:
Agree. Let's follow up with this in a subsequent PR. I'm also going to poke at the implementation in `ember-template-lint`, which I think will help inform the direction (as I'll need to be able to provide a custom storage directory from that tool).